src/x509_str.c

```diff
@@ -980,6 +982,9 @@ int wolfSSL_X509_STORE_set_flags(WOLFSSL_X509_STORE* store, unsigned long flag)
     if ((flag & WOLFSSL_CRL_CHECKALL) || (flag & WOLFSSL_CRL_CHECK)) {
         ret = wolfSSL_CertManagerEnableCRL(store->cm, (int)flag);
     }
+    else if (flag == 0) {
+        ret = wolfSSL_CertManagerDisableCRL(store->cm);
+    }
```

Comment:
I disagree with this.  From the OpenSSL documentation, X509_STORE_set_flags is ORing with the current flags, it is not overriding the flags, so a 0 is not disabling anything.  Disabling the CRL would come from clear_flags.

Reply:
> I disagree with this. From the OpenSSL documentation, X509_STORE_set_flags is ORing with the current flags, it is not overriding the flags, so a 0 is not disabling anything. Disabling the CRL would come from clear_flags.

@kareem-wolfssl, Thanks and your point is valid. However, wolfSSL activates CRL by default through the `wolfSSL_CertManagerEnableCRL() `function (i.e., `cm->crlEnabled `is always set to 1), and we conduct a CRL check in internal.c during state transition. This is not how OpenSSL operates.  

The OpenSSL documentation states:

...
<`X509_V_FLAG_POLICY_CHECK`> enables certificate policy checking, by default
no policy checking is performed. Additional information is sent to the
verification callback relating to policy checking.
...


Based on my current understanding, this flag is activated by the following OpenSSL functions which are absent in wolfSSL:
```
X509_VERIFY_PARAM_set1_policies()
opt_verify()
```

Hence, I suggest using the wolfSSL_X509_STORE_set_flags() to more closely align with the OpenSSL high-level behavior. Otherwise, we'd have to overhaul the X509 store code and implement additional features, which would be quite burdensome. Could you propose a solution that would be less troublesome?

Reply:
If OpenSSL is not enabling CRL by default, we should not be either when running in OpenSSL compatibility mode.  We could gate enabling CRL by default behind !OPENSSL_COMPATIBLE_DEFAULTS.
My fear is that a third-party application will call X509_STORE_set_flags(0), and will disable CRL without expecting to be disabled.

Reply:
@kareem-wolfssl, I see your point and updated `wolfSSL_X509_STORE_set_flags`().

Reply:
That isn't what I meant.  OpenSSL does not disable CRL when calling X509_STORE_set_flags(0), so we shouldn't either.  If the concern is us enabling CRL by default, you can gate us enabling CRL by default behind !OPENSSL_COMPATIBLE_DEFAULTS.  But X509_STORE_set_flags should not disable the CRL.  I am still not clear why we need to do this.

Reply:
@kareem-wolfssl,  I believe I've previously clarified why we require the modification to ensure better alignment between openssl and wolfssl at a higher level (beyond the API level, which, as you've correctly indicated, is an anticipated difference we need to accept).

Reply:
Discussed further over slack.  Haproxy is directly calling X509_STORE_set_flags with flags set to 0 which I was not aware of previously.  This change looks good to me.